### PR TITLE
Change `format` from enums to string type

### PR DESCRIPTION
**Pull request by sndrs**
_this is recreation of https://github.com/guardian/libs/pull/376_

## What does this change?

This changes the format types from enums to string.

## Why?

This helps bring DCR in line with the Frontend string properties reducing the number of transformations.

Secondly this will overcome difficulties of working with enums such as here:

https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/scripts/gen-stories/get-stories.js#L119

### CODE REVIEW

File: libs/@guardian/libs/src/format/ArticleDesign.ts [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
+	'DesignInterview',
+	'DesignEditorial',
+	'DesignQuiz',
+	'DesignInteractive',
+	'DesignPhotoEssay',
+	'DesignPrintShop',
+	'DesignObituary',
+	'DesignCorrection',
+	'DesignFullPageInteractive',
+	'DesignNewsletterSignup',
```
Comment (JamieB-gu) [inline libs/@guardian/libs/src/format/ArticleDesign.ts:22]:
```suggestion
	'Standard',
	'Gallery',
	'Audio',
	'Video',
	'Review',
	'Analysis',
	'Comment',
	'Letter',
	'Feature',
	'LiveBlog',
	'DeadBlog',
	'Recipe',
	'MatchReport',
	'Interview',
	'Editorial',
	'Quiz',
	'Interactive',
	'PhotoEssay',
	'PrintShop',
	'Obituary',
	'Correction',
	'FullPageInteractive',
	'NewsletterSignup',
```

File: libs/@guardian/libs/src/format/ArticleDisplay.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
-	Standard,
-	Immersive,
-	Showcase,
-	NumberedList,
-}
+const ArticleDisplays = [
+	'DisplayStandard',
+	'DisplayImmersive',
+	'DisplayShowcase',
+	'DisplayNumberedList',
```
Comment (JamieB-gu) [inline libs/@guardian/libs/src/format/ArticleDisplay.ts:5]:
```suggestion
	'Standard',
	'Immersive',
	'Showcase',
	'NumberedList',
```

File: libs/@guardian/libs/src/format/ArticlePillar.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
-	Sport = 2,
-	Culture = 3,
-	Lifestyle = 4,
-}
+const ArticlePillars = [
+	'PillarNews',
+	'PillarOpinion',
+	'PillarSport',
+	'PillarCulture',
+	'PillarLifestyle',
```
Comment (JamieB-gu) [inline libs/@guardian/libs/src/format/ArticlePillar.ts:6]:
```suggestion
	'News',
	'Opinion',
	'Sport',
	'Culture',
	'Lifestyle',
```

File: libs/@guardian/libs/src/format/ArticleSpecial.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-export enum ArticleSpecial {
-	SpecialReport = 5,
-	Labs = 6,
-}
+const ArticleSpecials = ['SpecialReport', 'SpecialLabs'] as const;
```
Comment (JamieB-gu) [inline libs/@guardian/libs/src/format/ArticleSpecial.ts:1]:
```suggestion
const ArticleSpecials = ['SpecialReport', 'Labs'] as const;
```

**Review by JamieB-gu — COMMENTED** [review]
Just re-creating the suggestions from the previous PR:

> Also, could I suggest dropping the Design, Display, Pillar and Special prefixes on each of the values? So we have something like:
>
> ```ts
> type ArticleDesign =
>  | 'Standard'
>  | 'Gallery'
>  | 'Review'
>  ...
> ```
>
> and so on. I think we already know this information from the name of the type itself, and it would probably make these easier to read and reduce string size in the bundle?

Referring to the 'Media' `Design`:

> I think this would be "Gallery", "Audio" and "Video" rather than "Media"?

I've added specific code suggestions for each of these too.

**Comment by sndrs** [comment]
I know it wasn't part of the original PR because the old repo didn't use [changesets](https://github.com/changesets/changesets), but it would be great to [fill out the changeset](https://github.com/guardian/csnx/edit/format-types/.changeset/eight-mayflies-yell.md?pr=%2Fguardian%2Fcsnx%2Fpull%2F67) with some more info if possible e.g. what a consumer will need to do etc

**Pull request closed without merging**